javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -280,8 +290,13 @@ object Integer {
 
   // Intrinsic, fallback on actual code for non-literal in JS
   @inline def numberOfLeadingZeros(i: scala.Int): scala.Int = {
-    if (LinkingInfo.esVersion >= ESVersion.ES2015) js.Math.clz32(i)
-    else clz32Dynamic(i)
+    linkTimeIf(LinkingInfo.targetPureWasm) {
+      // TODO: implement for Wasm without intrinsic?
+      throw new AssertionError("Not implemented.")
```

Comment:
You can use the algorithm under "See Hacker's Delight" from `clz32Dynamic` just below.